### DISCUSSION TITLE
Mount Examples - Repository

### DISCUSSION THREAD

**Discussion by hasansezertasan** [abersheeran/a2wsgi#38] [Show and tell]
Hello everyone, hope you're doing well. 😊

I'd like to introduce a small project I've been playing with. 🌱 [a2wsgi-examples](https://github.com/hasansezertasan/a2wsgi-examples)🚀 has a collection of examples showing how to mount applications built with ASGI/WSGI web frameworks to a FastAPI application.

I successfully mounted applications built with these frameworks: FastAPI, Starlette, Flask, Django, Bottle, Litestar, GradIO, PyWebIO and Flet. 💪

I failed attempting to mount applications built with these frameworks: Quart, Robyn, Falcon, Hug, Tornado, aiohttp... 😢

Check out the repository for more information 🤓.

---

A little backstory: I'm fascinated by Starlette and its features. I met with it this year thanks to FastAPI. I was planning to switch from Flask to FastAPI and I had an Admin Panel built with Flask Admin. I discovered this feature of Starlette and just mounted it into a dummy application, it worked! At first, I was using the same routes from Flask, but slowly I rewrote those with FastAPI. The day came, and I finished the migration. I can say that it indeed helped me a lot. Switching from a backend that has +20K lines of code was intimidating, but with less than 10 lines of code, I was able to start my migration process. I started migration and line by line, here I am! 🚀 (My code base is now less than 15K)

**Comment by abersheeran** [comment]
Great work, but I remember that aiohttp does not support ASGI or WSGI protocols, and it may not be mounted under fastapi now.

**Reply by hasansezertasan** [reply]
Thank you @abersheeran

Oh I see, I'm trying to update the project as much as I can. I was not able to figure out why was aiohttp or other frameworks I failed to mount. Thank you for the hint.

Edit: I just thought about something, I can't see any examples in the repository. Are you open to idea of adding examples? I can work on it.

**Reply by abersheeran** [reply]
> Edit: I just thought about something, I can't see any examples in the repository. Are you open to idea of adding examples? I can work on it.

Which type of example are you referring to?

**Reply by hasansezertasan** [reply]
Can be something like the examples from the repo I shared here. One example can show how to mount a Flask application to a Starlette application using a2wsgi, things like that.

And a compatibility list to inform people which frameworks can be used with a2wsgi could be quite good, wdyt?

**Reply by abersheeran** [reply]
A compatibility list sounds good, I'll add it soon. PR is also welcome.

**Reply by hasansezertasan** [reply]
@abersheeran The Compatibility list looks quite good, well done! 👏
